### ISSUE TITLE
Add custom error type for broadcasting

### ISSUE THREAD

**Issue by catrias** [cosmos/cosmjs#1096]
Stargate client `broadcastTx` method only provides text on error.
```
throw new Error(
  `Broadcasting transaction failed with code ${broadcasted.code} (codespace: ${broadcasted.codeSpace}). Log: ${broadcasted.log}`,
);
```

It would be useful to use some custom error class to include the code like
```
export class BroadcastTxError extends Error {
  constructor(public readonly code: number, message?: string) {
    super(message);
  }
}

throw new BroadcastTxError(
  broadcasted.code
  `Broadcasting transaction failed with code ${broadcasted.code} (codespace: ${broadcasted.codeSpace}). Log: ${broadcasted.log}`,
);
```

**Comment by webmaster128** [comment]
Good idea, thanks. See #1235

**Comment by webmaster128** [comment]
Shipped as part of CosmJS 0.29.0
